### DISCUSSION TITLE
how to predefine signals for docker compose run

### DISCUSSION THREAD

**Discussion by denis-peshkov** [datalust/seq-tickets#2226] [Help]
As a developer I added seq docker image into my project, I'd like some how to configure seq signals on first run
so in case I delete the volume and start seq from scratch it will contain not default two signals but a lot I would configure

**Comment by liammclennan** [comment]
Hi @denis-peshkov 

You can also use a [bind mound](https://docs.docker.com/storage/bind-mounts/) to keep your Seq data outside of docker.

You could also script it with seqcli. See https://docs.datalust.co/docs/command-line-client#signal-list, https://docs.datalust.co/docs/command-line-client#signal-list, https://docs.datalust.co/docs/command-line-client#template-export and https://docs.datalust.co/docs/command-line-client#template-import.

**Reply by denis-peshkov** [reply]
Sorry, I don't get how to configure the file 

Is it possible to get syntax for the seed file ?
What exactly file I have to replace?

I have no possibility to run external client, only dockerfile

**Reply by nblumhardt** [reply]
Thanks for your reply, Denis.

It's not possible to initialize signals/dashboards etc. without running some CLI commands. The `seqcli` template system is introduced in https://blog.datalust.co/copying-signals-and-dashboards-using-seqcli-templates/, which is probably the best starting point.

Running `seqcli help template export` and `seqcli help template import` should give you the full details of how those commands work; running an `export` and then viewing the files will show the format.

Perhaps reconsidering why the volume would need to be deleted/started from scratch is worthwhile, though? Retention policies (Data > Storage) enable deleting log/trace data after a fixed time interval, for example.

Hope this helps,
Nick
